src/timeMachine/components/dynamicFluxFunctionsToolbar/TransformToolbarFunctions.tsx

```diff
@@ -1,23 +1,38 @@
 // Libraries
-import React, {FC, ReactElement} from 'react'
-
+import React, {FC, ReactElement, useMemo} from 'react'
 // Components
 import {EmptyState, ComponentSize} from '@influxdata/clockface'
 
 // Types
-import {FluxToolbarFunction} from 'src/types/shared'
+import {Fluxdocs} from 'src/client/fluxdocsdRoutes'
 
 interface Props {
-  funcs: FluxToolbarFunction[]
+  funcs: any
   searchTerm?: string
-  children: (funcs: FluxToolbarFunction[]) => JSX.Element | JSX.Element[]
+  children: (funcs: Fluxdocs[]) => JSX.Element | JSX.Element[]
 }
 
 const TransformToolbarFunctions: FC<Props> = props => {
   const {searchTerm, funcs, children} = props
 
-  const filteredFunctions = funcs.filter(func =>
-    func.name.toLowerCase().includes(searchTerm.toLowerCase())
+  const sortedFunctions = funcs.sort((a, b) => {
+    if (a.package.toLowerCase() === b.package.toLowerCase()) {
+      return a.name.toLowerCase() < b.name.toLowerCase() ? -1 : 1
+    } else {
+      return a.package.toLowerCase() < b.package.toLowerCase() ? -1 : 1
+    }
+  })
+
+  const filteredFunctions = useMemo(
+    () =>
+      sortedFunctions.filter(fn => {
+        return (
+          !searchTerm.length ||
+          fn.name.toLowerCase().includes(searchTerm.toLowerCase()) ||
+          fn.package.toLowerCase().includes(searchTerm.toLowerCase())
+        )
+      }),
+    [searchTerm, sortedFunctions]
```

Comment:
This is not quite right. I still see a lot of unnecessary re-sorting.

You want to apply the `useMemo` on the `sortedFunctions` which is an array. Basically, you want the same sorted array unless new functions are added for some reason.

`filteredFunctions` does not need the `useMemo` I don't think. You can double check this after making the change.